www/js/controllers.js

```diff
@@ -55,11 +55,29 @@ angular.module('starter.controllers', [])
     }
   }
 
-  $scope.addToRecentSearches = function() {
-    RecentSearches.push($scope.searchText);
-    Store.set('sikher_recent', RecentSearches);
-  }
-
+	$scope.addToRecentSearches = function () {
+		//remove from recent search if it was same as new one
+		for (var i = 0; i < RecentSearches.length; i++) {
+			if (RecentSearches[i].searchText === $scope.searchText
+				&& RecentSearches[i].searchType === $scope.search
+				) {
+				RecentSearches.splice(i, 1);
+			}
+		}
+
+		RecentSearches.push({
+			searchText: $scope.searchText,
+			searchType: $scope.search
+		});
+		Store.set('sikher_recent', RecentSearches);
+	};
+
+	$scope.getRecentSearchClass = function (searchType) {
+		if (searchType === 'gurmukhi' || searchType === 'gurmukhi_search') {
```

Comment:
Finally, this should really use the `$filter('placeholder')`, which already has the necessary mapping between searchType and class